src/fluvio.test.ts

```diff
@@ -124,6 +125,58 @@ describe('Fluvio Batch Producer', () => {
     })
 })
 
+describe('Configures a SmartModule with a Filter configuration', () => {
+    jest.setTimeout(100000) // 100 seconds
+    let admin: FluvioAdmin
+    let fluvio: Fluvio
+    let topic: string
+
+    beforeAll(async () => {
+        topic = uuidV4()
+        fluvio = await Fluvio.connect()
+        admin = await fluvio.admin()
+
+        console.log(`Creating topic ${topic}`)
+
+        await admin.createTopic(topic)
+        await sleep(topic_create_timeout)
+    });
+
+    afterAll(async () => {
+        console.log(`Deleting topic ${topic}`)
+        await admin.deleteTopic(topic)
+        await sleep(topic_create_timeout)
+    });
+
+    test('Applies a Filter SmartModule on the provided stream', async () => {
+        const producer = await fluvio.topicProducer(topic)
+        const consumer = await fluvio.partitionConsumer(topic, 0);
+        const wasmSmartModule = await fs.promises.readFile('./fixtures/server_logs_filter.wasm.gz');
+        const serverLogsFile = await fs.promises.readFile('./fixtures/server_log.json', 'utf8');
+        const serverLogs: { message: string; level: string; }[] = JSON.parse(serverLogsFile);
+        const stream = await consumer.streamWithConfig(Offset.FromBeginning(), {
+            smartmoduleType: SmartModuleType.Filter,
+            smartmoduleData: wasmSmartModule.toString('base64'),
+        });
+        const receivedLogs = [];
+
+        for (let log of serverLogs) {
+            producer.send(uuidV4(), JSON.stringify(log));
+        }
+
+        for await (const record of stream) {
+            receivedLogs.push(JSON.parse(record.valueString()));
+
+            if (receivedLogs.length >= 5) {
+                break;
+            }
+        }
+
+        expect(receivedLogs.find((log) => log.level === 'debug')).toBeUndefined();
+        expect(receivedLogs.length).toBe(serverLogs.filter((log) => log.level !== 'debug').length);
```

Comment:
Here Im using the filter smart module to replicate the same behavior from the tutorial but using the NodeJS client.
Basically I want to filter debug logs available in the `server_log.json` file.